src/equations/acoustic_perturbation_2d.jl

```diff
@@ -0,0 +1,163 @@
+[`source_terms_convergence_test`](@ref).
+"""
+function initial_condition_convergence_test(x, t, equations::AcousticPerturbationEquations2D)
+  c = 2.0
+  A = 0.2
+  L = 2.0
+  f = 2.0 / L
+  a = 1.0
+  omega = 2 * pi * f
+  init = c + A * sin(omega * (x[1] + x[2] - a*t))
+
+  v1_prime = init
+  v2_prime = init
+  p = init * init
+
+  return SVector(v1_prime, v2_prime, p)
+end
+
+"""
+  source_terms_convergence_test(u, x, t, equations::AcousticPerturbationEquations2D)
+
+Source terms used for convergence tests in combination with
+[`initial_condition_convergence_test`](@ref).
+"""
+function source_terms_convergence_test(u, x, t, equations::AcousticPerturbationEquations2D)
+  @unpack v_mean, rho_mean, c_mean = equations
+
+  c = 2.0
+  A = 0.2
+  L = 2.0
+  f = 2.0 / L
+  a = 1.0
+  omega = 2 * pi * f
+
+  si, co = sincos(omega * (x[1] + x[2] - a * t))
+  tmp = v_mean[1] + v_mean[2] - a
+
+  du1 = du2 = A * omega * co * (2 * c + tmp + 2/rho_mean * A * si)
+  du3 = A * omega * co * (2 * c_mean^2 * rho_mean + 2 * c * tmp + 2 * A * tmp * si)
+
+  return SVector(du1, du2, du3)
+end
+
+
+# Calculate 1D flux for a single point
+@inline function calcflux(u, orientation, equations::AcousticPerturbationEquations2D)
+  v1_prime, v2_prime, p_prime = u
+  @unpack v_mean, rho_mean, c_mean = equations
+
+  if orientation == 1
+    f1 = v_mean[1] * v1_prime + v_mean[2] * v2_prime + p_prime / rho_mean
+    f2 = zero(eltype(u))
+    f3 = c_mean^2 * rho_mean * v1_prime + v_mean[orientation] * p_prime
+  else
+    f1 = zero(eltype(u))
+    f2 = v_mean[1] * v1_prime + v_mean[2] * v2_prime + p_prime / rho_mean
+    f3 = c_mean^2 * rho_mean * v2_prime + v_mean[orientation] * p_prime
+  end
+
+  return SVector(f1, f2, f3)
+end
+
+
+function flux_lax_friedrichs(u_ll, u_rr, orientation, equations::AcousticPerturbationEquations2D)
+  f_ll = calcflux(u_ll, orientation, equations)
+  f_rr = calcflux(u_rr, orientation, equations)
+
+  v_ll = u_ll[orientation] + equations.v_mean[orientation]
+  v_rr = u_rr[orientation] + equations.v_mean[orientation]
+  c0 = equations.c_mean
+  speed = max(abs(v_ll), abs(v_rr)) + c0
+
+  return 0.5 * ( (f_ll + f_rr) - speed * (u_rr - u_ll) )
+end
+
+
+@inline have_constant_speed(::AcousticPerturbationEquations2D) = Val(true)
+
+@inline function max_abs_speeds(equations::AcousticPerturbationEquations2D)
+  return abs.(equations.v_mean) .+ equations.c_mean
+end
+
+
+# Convert conservative variables to primitive
+@inline cons2prim(u, equations::AcousticPerturbationEquations2D) = u
+
+# Convert conservative variables to entropy variables
+@inline cons2entropy(u, equations::AcousticPerturbationEquations2D) = u
+
+
+# Calculate entropy for a conservative state `cons`
+@inline entropy(u, equations::AcousticPerturbationEquations2D) = entropy(u[1], equations)
+
+
+# Calculate total energy for a conservative state `cons`
+@inline energy_total(u, equations::AcousticPerturbationEquations2D) = energy_total(u[1], equations)
```

Comment:
```suggestion
```
That will probably yield a stack overflow and is not tested, I guess. If you want to include something like this, it should be fixed and tested.

Reply:
You're right. @lchristm If the tests pass with those lines removed, then just take them out for now until we have a need for them.

Reply:
Done in 284d1ecce8a5ed2158fb767b1613812bf571f5db